test/src/unit-capi-config.cc

```diff
@@ -230,7 +230,7 @@ void check_save_to_file() {
   ss << "rest.curl.buffer_size 524288\n";
   ss << "rest.curl.verbose false\n";
   ss << "rest.http_compressor any\n";
-  ss << "rest.load_enumerations_on_array_open true\n";
+  ss << "rest.load_enumerations_on_array_open false\n";
   ss << "rest.load_metadata_on_array_open true\n";
   ss << "rest.load_non_empty_domain_on_array_open true\n";
```

Comment:
Why are these options restricted to REST only? Especially the first two have use cases outside of it.

Reply:
Yeah also crossed my mind while working on this.. I didn't introduce any new config options in this PR but maybe `sm.load_X_on_array_open` or `sm.array_open.load_X` would be more fitting for these? I think they were each introduced to tune REST behavior and that's probably where the names originated.

I opened [SC-53109](https://app.shortcut.com/tiledb-inc/story/53109/consider-renaming-rest-load-x-on-array-open-config-options) to consider renaming them, probably best to handle the renaming separately.